### PR TITLE
storage: refactor check_txn_status_missing_lock

### PR DESCRIPTION
**Pull request by ekexium**
### What problem does this PR solve?

check_txn_status_missing_lock takes two booleans, which is an
anti-pattern.

### What is changed and how it works?

Use an enum instead.

### Check List 

Tests 

- No code

### Release note

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -945,7 +962,14 @@ impl<S: Snapshot> MvccTxn<S> {
             // The rollback must be protected, see more on
             // [issue #7364](https://github.com/tikv/tikv/issues/7364)
             _ => self
-                .check_txn_status_missing_lock(primary_key, rollback_if_not_exist, true)
+                .check_txn_status_missing_lock(
+                    primary_key,
+                    if rollback_if_not_exist {
```
Comment (nrc) [inline src/storage/mvcc/txn.rs:967]: And this

File: src/storage/mvcc/txn.rs [hunk 2]
```diff
@@ -728,32 +752,40 @@ impl<S: Snapshot> MvccTxn<S> {
...
+                let write = self.get_write(action, ts);
+                self.put_write(primary_key, ts, write.as_ref().to_bytes());
+                MVCC_CHECK_TXN_STATUS_COUNTER_VEC.rollback.inc();
+
+                Ok(TxnStatus::LockNotExist)
             }
         }
     }
 
+    fn get_write(&self, action: MissingLockAction, ts: TimeStamp) -> Write {
```
Comment (youjiali1995) [inline src/storage/mvcc/txn.rs:781]: It's not a proper name and I can't think of one. How about adding a method to `MissingLockAction`:
```rust
impl MissingLockAction {
...
    fn into_write/or rollback(self, ts: TimeStamp) -> Write {
        match self {
            MissingLockAction::Rollback => Write::new_rollback(ts, false),
            MissingLockAction::ProtectedRollback => Write::new_rollback(ts, true),
            _ => unreachable!(),
        }
    }
}

**Comment by nrc** [comment]
@youjiali1995 comment addressed, please take another look

**Pull request merged**